### PR TITLE
Dnn_ Net_ A bug occurred when compiling getUnconnectedOutLayersNames into JavaScript

### PR DESCRIPTION
**Pull request by laolaolulu**
ret_type = re.sub('(^|[^\w])' + key + '($|[^\w])', type_dict[key], ret_type)
When compiling JavaScript, this code converts vector<std:: string>to vector std:: string
Details: https://laolaolulu.github.io/posts/emscripten-opencv/#opencvjs%E7%BC%96%E8%AF%91getunconnectedoutlayersnames%E8%BF%87%E7%A8%8B%E4%B8%AD%E9%94%99%E8%AF%AF%E5%A4%84%E7%90%86

```
force_builders=Custom
build_image:Docs=docs-js:18.04
build_image:Custom=javascript
buildworker:Custom=linux-1,linux-4,linux-f1
```

### CODE REVIEW

**Comment by asmorkalov** [comment]
```
Run cd $HOME/build
Scanning dependencies of target check_flake8
Running flake8
./modules/js/generator/embindgen.py:485:69: E999 TabError: inconsistent use of tabs and spaces in indentation
	    ret_type = re.sub(r"\b" + key + r"\b", type_dict[key], ret_type)
                                                                    ^
1     E999 TabError: inconsistent use of tabs and spaces in indentation
1
make[3]: *** [CMakeFiles/check_flake8.dir/build.make:58: CMakeFiles/check_flake8] Error 1
```

**Comment by asmorkalov** [comment]
```
modules/js/src/core_bindings.cpp:463: trailing whitespace.
+    register_vector<double>("DoubleVector"); 
```

**Comment by asmorkalov** [comment]
@laolaolulu please revert formatting changes. Pure code re-formatting does not bring value, but makes review harder, merge conflicts in 4.x->5.x merges and other issues for maintainers.

**Comment by laolaolulu** [comment]
> @laolaolulu please revert formatting changes. Pure code re-formatting does not bring value, but makes review harder, merge conflicts in 4.x->5.x merges and other issues for maintainers.

What formatting code plugin should I use for the vscode I am using?

**Comment by asmorkalov** [comment]
Wiki page on coding style and formatting: https://github.com/opencv/opencv/wiki/Coding_Style_Guide
We recommend you the codding style above and .clang-format attached to the wiki page, but for new code only. Please do not touch not relevant pieces.

File: platforms/js/build_js.py [hunk 1]
```diff
@@ -256,6 +256,9 @@ def build_loader(self):
                         help="Specify configuration file with own list of exported into JS functions")
     parser.add_argument('--webnn', action="store_true", help="Enable WebNN Backend")
 
+    parser.add_argument("-DCMAKE_TOOLCHAIN_FILE", default="", help="DCMAKE_TOOLCHAIN_FILE")
+    parser.add_argument("-DCMAKE_CROSSCOMPILING_EMULATOR",default="",help="DCMAKE_CROSSCOMPILING_EMULATOR")
+
```
Comment (asmorkalov) [inline platforms/js/build_js.py:261]: The new parameters are not used in the script.

**Review by asmorkalov — CHANGES_REQUESTED** [review]
Changes in opencv_js.config.py are missing.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**